### PR TITLE
Add citation functionality to the SecHud

### PR DESCRIPTION
**Pull request by Dingo-Dongler**
## About The Pull Request

Allows people wearing the sechud to give people citations on examine, similar to changing criminal status and adding crimes.

## Why It's Good For The Game

Citations might actually be used if sec didn't have to walk all the way to a console to give people littering fines.

## Changelog
:cl:
add: Examining people with sechuds will let you give someone an on the spot citation.
/:cl:

### CODE REVIEW

**Comment by trollbreeder** [comment]
Pick up that can

**Comment by Anonmare** [comment]
[Citation needed]

**Comment by AnonymousNow** [comment]
Please add citation summary at the end of the round, so that we can see everything appended to the record of that one assistant that never broke the law, but still warranted 73 citations for borderline non-offences.

**Comment by trollbreeder** [comment]
> Please add citation summary at the end of the round, so that we can see everything appended to the record of that one assistant that never broke the law, but still warranted 73 citations for borderline non-offences.

Guess that means we're Goonstation now.

**Comment by nfreader** [comment]
Oh thank you! I meant to add this in a later PR but it stale'd out. 

@AnonymousNow You can see citations from a round in records.html under `parsed-logs` or on Statbus. I'm still evaluating how to handle citations in the DB :)

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -431,6 +431,38 @@
...
+							"job" = "Citation Server",
+							"message" = message,
+							"targets" = list("[P.owner] ([P.ownjob])"),
+							"automated" = 1
+						))
+						signal.send_to_receivers()
+						usr.log_message("(PDA: Citation Server) sent \"[message]\" to [signal.format_target()]", LOG_PDA)
+				GLOB.data_core.addCitation(R.fields["id"], crime)
+				investigate_log("New Citation: <strong>[t1]</strong> Fine: [fine] | Added to [R.fields["name"]] by [key_name(usr)]", INVESTIGATE_RECORDS)
+			return
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/human/human.dm]: down one?

Comment (Dingo-Dongler) [reply code/modules/mob/living/carbon/human/human.dm]: what?

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -431,6 +431,38 @@
...
+				if(!HAS_TRAIT(H, TRAIT_SECURITY_HUD))
+					return
+				var/t1 = stripped_input("Please input citation crime:", "Security HUD", "", null)
+				var/fine = FLOOR(input("Please input citation fine:", "Security HUD", 50) as num|null, 1)
+				if(isnull(fine))
+					return
+				if(fine < 0)
+					to_chat(usr, "<span class='warning'>You're pretty sure that's not how money works.</span>")
+					return
+				fine = min(fine, 1000) //maxFine in the regular sec thing. pretty sure this should be a referece to that somehow.
```
Comment (Dingo-Dongler) [inline code/modules/mob/living/carbon/human/human.dm]: maxFine in the security console doesn't _feel_ like it should be a local variable, should it be defined as one of those static global variables like MAX_FINE?

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (optimumtact) [reply code/modules/mob/living/carbon/human/human.dm]: @ShizCalev

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (Dingo-Dongler) [reply code/modules/mob/living/carbon/human/human.dm]: How would people feel about this being in server config? Unnecessary or justified?

Comment (optimumtact) [reply code/modules/mob/living/carbon/human/human.dm]: if you want to go to the extra effort, I won't stop you

Comment (Dingo-Dongler) [reply code/modules/mob/living/carbon/human/human.dm]: epic

File: code/game/machinery/computer/security.dm [hunk 3]
```diff
@@ -740,6 +739,7 @@ What a mess.*/
 								to_chat(usr, "<span class='warning'>You're pretty sure that's not how money works.</span>")
```
Comment (Cyberboss) [inline code/game/machinery/computer/security.dm]: Line 733 should mention the max fine in the prompt.

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -431,6 +431,38 @@
...
 				return
 
+			if(href_list["add_citation"])
+				if(!H.canUseHUD())
+					return
+				if(!HAS_TRAIT(H, TRAIT_SECURITY_HUD))
+					return
+				var/t1 = stripped_input("Please input citation crime:", "Security HUD", "", null)
+				var/fine = FLOOR(input("Please input citation fine:", "Security HUD", 50) as num|null, 1)
+				if(isnull(fine))
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/human.dm]: You need to check for the presence of the hud between and after the prompts. Also mention the max fine.

File: code/modules/mob/living/carbon/human/human.dm [hunk 5]
```diff
@@ -431,6 +431,38 @@
...
+							"job" = "Citation Server",
+							"message" = message,
+							"targets" = list("[P.owner] ([P.ownjob])"),
+							"automated" = 1
+						))
+						signal.send_to_receivers()
+						usr.log_message("(PDA: Citation Server) sent \"[message]\" to [signal.format_target()]", LOG_PDA)
+				GLOB.data_core.addCitation(R.fields["id"], crime)
+				investigate_log("New Citation: <strong>[t1]</strong> Fine: [fine] | Added to [R.fields["name"]] by [key_name(usr)]", INVESTIGATE_RECORDS)
+			return
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/human.dm]: I believe the correction is
```suggestion
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human.dm (hunk 4 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/human/human.dm]: this needs to be addressed

> You need to check for the presence of the hud between and after the prompts.

**Comment by ExcessiveUseOfCobblestone** [comment]
need to check for hud usage before and after the inputs

**Comment by ExcessiveUseOfCobblestone** [comment]
oh wait old code hehe

**Pull request merged**